src/Microsoft.AspNetCore.Server.Kestrel.Core/Adapter/Internal/AdaptedPipeline.cs

```diff
@@ -111,8 +117,8 @@ public class AdaptedPipeline
             {
                 if (stream == null)
                 {
-                    // If the stream is null then we're going to abort the connection
-                    throw new ConnectionAbortedException();
+                    // REVIEW: Do we need an exception here?
+                    return;
```

Comment:
Instead of throwing here, it should be the responsibility of ReadInputAsync's caller to not call it with an empty stream. That means the caller would also need to call `Input.Writer.Complete(Exception)` and `_transportInputPipeReader.Complete();`.